### PR TITLE
Fix masking order

### PR DESCRIPTION
**Pull request by droidicus**
This fixes the same issue as #14831 but for the BatchNormalization layer and Bidirectional wrapper.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: It has been 14 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by fchollet** [comment]
Thanks for the PR, and sorry for the delayed review. Could you please rebase, since the layers have changed recently? I believe the fix is no longer necessary in BN but may be necessary in `Bidirectional`.

**Comment by droidicus** [comment]
Rebased, and should be ready to go.

**Review by fchollet — DISMISSED** [review]
LGTM, thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by fchollet** [comment]
I think we can merge this when the tests pass.

**Comment by droidicus** [comment]
The master this was based on previously was causing the test error, rebased to current master to (hopefully) fix the test...

**Review by fchollet — APPROVED** [review]

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 18 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by fchollet** [comment]
Sorry for the delay in getting this tested. Could you please rebase to fix the merge conflicts?

**Comment by drpngx** [comment]
@droidicus any chance to pull rebase and push again?

**Comment by droidicus** [comment]
It looks like the branch is in conflict due to the bug being fixed upstream already. This PR can be closed.
